src/libultra/io/vimodepallan1.c

```diff
@@ -1,32 +1,46 @@
+/**
+ * @file vimodepallan1.c
+ *
+ * PAL LAN1 Video Mode
+ *
+ * L = Low Resolution
+ * A = Anti-Aliased
+ * N = Deinterlaced
+ * 1 = 16-bit Framebuffer
+ */
 #include "global.h"
+#include "ultra64/viint.h"
 
 OSViMode osViModePalLan1 = {
-    0x10, // type
+    OS_VI_PAL_LAN1, // type
     {
         // comRegs
-        0x311E,       // ctrl
-        SCREEN_WIDTH, // width
-        0x4541E3A,    // burst
-        0x271,        // vSync
-        0x170C69,     // hSync
-        0xC6F0C6D,    // leap
-        0x800300,     // hStart
-        0x200,        // xScale
-        0             // vCurrent
+        VI_CTRL_TYPE_16 | VI_CTRL_GAMMA_DITHER_ON | VI_CTRL_GAMMA_ON | VI_CTRL_DIVOT_ON | VI_CTRL_ANTIALIAS_MODE_1 |
+            VI_CTRL_PIXEL_ADV_3, // ctrl
+        WIDTH(320),              // width
+        BURST(58, 30, 4, 69),    // burst
+        VSYNC(625),              // vSync
+        HSYNC(3177, 23),         // hSync
+        LEAP(3183, 3181),        // leap
+        HSTART(128, 768),        // hStart
+        SCALE(2, 0),             // xScale
+        VCURRENT(0),             // vCurrent
     },
     { // fldRegs
       {
-          0x280,    // origin
-          0x400,    // yScale
-          0x5F0239, // vStart
-          0x9026B,  // vBurst
-          2,        // vIntr
+          // [0]
+          ORIGIN(640),         // origin
```

Comment:
based on usage in `__osViSwapContext` this isn't directly the origin reg value (thankfully), merely an offset (which I have no idea what its purpose is), so it probably shouldn't use `ORIGIN()`

also inspired from `ViMode_Configure`, 640 is 320\*2 (width\*2) ?

Reply:
`ORIGIN()` is for `OSViFieldRegs::origin`, not `VI_ORIGIN_REG`

If we wanted we could make `ORIGIN` be a macro in terms of width, bitdepth and target resolution:
`#define ORIGIN(width, fb16Bit, loRes) ((width) * 2 * ((fb16Bit) ? 1 : 2) * ((loRes) ? 1 : 2)))`
If it were just in terms of `width` it would not work for other VI modes that are not included in oot, breaking compatibility. I'm not sure if it's worth the added complexity

Reply:
It would be good to comment on the macros what their purpose is then (if not use prefixes), since "origin" is also a vi reg and the macros look just like the vi reg macros family WIDTH, HSYNC etc